### PR TITLE
Raise hand feature react composite

### PR DESCRIPTION
**Pull request by rzdor**
# What
This is react composite changes for raise hand feature.
Latest available build is here https://hero-calling-app10.azurewebsites.net/
image
image

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/calling-component-bindings/src/callControlSelectors.ts [hunk 1]
```diff
@@ -140,6 +141,38 @@ export type ScreenShareButtonSelector = (
...
+ *
+ * @public
+ */
+export const raiseHandButtonSelector: RaiseHandButtonSelector = reselect.createSelector(
+  [getLocalParticipantRaisedHand, /* @conditional-compile-remove(PSTN-calls) */ getCallState],
+  (raisedHand, /* @conditional-compile-remove(PSTN-calls) */ callState) => {
+    return {
+      checked: raisedHand ? true : false,
+      /* @conditional-compile-remove(PSTN-calls) */
+      disabled: callState === 'InLobby' ? true : callState === 'Connecting' ?? false
```
Comment (JamesBurnside) [inline packages/calling-component-bindings/src/callControlSelectors.ts:171]: potentially onhold as well?

File: packages/calling-component-bindings/src/callControlSelectors.ts [hunk 2]
```diff
@@ -140,6 +141,38 @@ export type ScreenShareButtonSelector = (
...
+
+/* @conditional-compile-remove(raise-hand) */
+/**
+ * Selector for {@link RaiseHandButton} component.
+ *
+ * @public
+ */
+export const raiseHandButtonSelector: RaiseHandButtonSelector = reselect.createSelector(
+  [getLocalParticipantRaisedHand, /* @conditional-compile-remove(PSTN-calls) */ getCallState],
+  (raisedHand, /* @conditional-compile-remove(PSTN-calls) */ callState) => {
```
Comment (JamesBurnside) [inline packages/calling-component-bindings/src/callControlSelectors.ts:167]: these should be `(raise-hand` and not `PSTN-calls`

Comment (JamesBurnside) [reply packages/calling-component-bindings/src/callControlSelectors.ts:167]: (in a couple places in this file)

File: packages/react-components/src/components/ParticipantList.tsx [hunk 3]
```diff
@@ -124,6 +124,7 @@ const onRenderParticipantDefault = (
   onParticipantClick?: (participant?: ParticipantListParticipant) => void,
   showParticipantOverflowTooltip?: boolean,
   participantAriaLabelledBy?: string,
+  /* @conditional-compile-remove(raise-hand) */
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ParticipantList.tsx:127]: presumably theme should not be conditionally compiled here?

File: packages/react-components/src/components/ParticipantList.tsx [hunk 4]
```diff
@@ -258,6 +278,9 @@ export const ParticipantList = (props: ParticipantListProps): JSX.Element => {
   const displayedParticipants: ParticipantListParticipant[] = useMemo(() => {
     return onRenderParticipant ? participants : getParticipantsForDefaultRender(participants, excludeMe, myUserId);
   }, [participants, excludeMe, myUserId, onRenderParticipant]);
+
+  sortParticipants(displayedParticipants as CallParticipantListParticipant[]);
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ParticipantList.tsx:282]: this is being cast as `CallParticipantListParticipant` but there's no guarantee that's correct.
Meaning in the sort function `a.raisedHand.raisedHandOrderPosition` could throw an exception if a.raisedHand is undefined

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 5]
```diff
@@ -191,6 +192,9 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
 
   const cameraButtonIsEnabled = isEnabled(options?.cameraButton);
 
+  /* @conditional-compile-remove(raise-hand) */
+  const raiseHandButtonIsEnabled = true;
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:196]: Should this use the `isEnabled` function to ensure the options are applied?

File: packages/react-composites/src/composites/CallComposite/components/buttons/RaiseHand.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  // The value of `CallControlOptions.raiseHandButton`.
+  option?: boolean | { disabled: boolean };
+  displayType?: CallControlDisplayType;
+  styles?: ControlBarButtonStyles;
+  disabled?: boolean;
+}): JSX.Element => {
+  const raiseHandButtonProps = usePropsFor(RaiseHandButton);
+  const styles = useMemo(() => concatButtonBaseStyles(props.styles ?? {}), [props.styles]);
+
+  const raiseHandButtonDisabled = (): boolean => {
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/components/buttons/RaiseHand.tsx:26]: why is this a function and not just `const raiseHandButtonDisabled = isDisabled(props.option);`

File: packages/react-composites/src/composites/common/types/CommonCallControlOptions.tsx [hunk 7]
```diff
@@ -77,6 +77,18 @@ export type CommonCallControlOptions = {
    * @defaultValue true
    */
   moreButton?: boolean;
+  /* @conditional-compile-remove(raise-hand) */
+  /**
+   * Show, Hide or Disable the screen share button during a call.
+   * @defaultValue true
+   */
+  raiseHandButton?: boolean | { disabled: boolean };
+  /* @conditional-compile-remove(PSTN-calls) */
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/common/types/CommonCallControlOptions.tsx:86]: these comments don't look intentional, copy-paste error?

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-videos-Mobile-Android-Landscape-linux.png
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-videos-Mobile-Android-Landscape-linux.png:1]: control bar is going into overflow

File: packages/calling-component-bindings/src/videoGallerySelector.ts [hunk 8]
```diff
@@ -89,12 +97,19 @@ export const videoGallerySelector: VideoGallerySelector = createSelector(
     const dominantSpeakersMap: Record<string, number> = {};
     dominantSpeakerIds?.forEach((speaker, idx) => (dominantSpeakersMap[speaker] = idx));
     const noRemoteParticipants = [];
+
+    let convert = convertRemoteParticipantToVideoGalleryRemoteParticipant as any;
```
Comment (dmceachernmsft) [inline packages/calling-component-bindings/src/videoGallerySelector.ts:101]: If we are doing this in the selector utils do we need to do this here too?

File: packages/react-composites/src/composites/common/types/CommonCallControlOptions.tsx (hunk 7 above)
Comment (rzdor) [reply packages/react-composites/src/composites/common/types/CommonCallControlOptions.tsx:86]: good catch, thanks

File: packages/calling-component-bindings/src/videoGallerySelector.ts (hunk 8 above)
Comment (rzdor) [reply packages/calling-component-bindings/src/videoGallerySelector.ts:101]: thanks, rewrote it

File: packages/react-composites/src/composites/CallComposite/components/buttons/RaiseHand.tsx (hunk 6 above)
Comment (rzdor) [reply packages/react-composites/src/composites/CallComposite/components/buttons/RaiseHand.tsx:26]: changed

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx (hunk 5 above)
Comment (rzdor) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:196]: Yes, thanks

File: packages/react-components/src/components/ParticipantList.tsx (hunk 4 above)
Comment (rzdor) [reply packages/react-components/src/components/ParticipantList.tsx:282]: Agree, moved type into sorting

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-videos-Mobile-Android-Landscape-linux.png
Comment (rzdor) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-videos-Mobile-Android-Landscape-linux.png:1]: for mobile view moved raise hand action under "..."

File: packages/calling-component-bindings/src/callControlSelectors.ts (hunk 1 above)
Comment (dmceachernmsft) [reply packages/calling-component-bindings/src/callControlSelectors.ts:169]: I agree this should probably be disabled in this state

File: packages/react-composites/src/composites/common/Drawer/MoreDrawer.tsx [hunk 9]
```diff
@@ -252,6 +257,30 @@ export const MoreDrawer = (props: MoreDrawerProps): JSX.Element => {
...
+        ? localeStrings.component.strings.raiseHandButton.onLabel
+        : localeStrings.component.strings.raiseHandButton.offLabel,
+      onItemClick: () => {
+        if (raiseHandButtonProps.onToggleRaiseHand) {
+          raiseHandButtonProps.onToggleRaiseHand();
+        }
+        onLightDismiss();
+      },
+      iconProps: {
+        iconName: raiseHandIconName,
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/common/Drawer/MoreDrawer.tsx:278]: nit: better name `raiseHandButtonIcon` we dont want to include the suffix `name` in the icons

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/LoadingVideo.test.ts-snapshots/horizontal-gallery-with-loading-spinners-Mobile-Android-Landscape-linux.png
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/LoadingVideo.test.ts-snapshots/horizontal-gallery-with-loading-spinners-Mobile-Android-Landscape-linux.png:1]: we should be hiding this in the more button in the legacy control bar experience as well

Comment (rzdor) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/LoadingVideo.test.ts-snapshots/horizontal-gallery-with-loading-spinners-Mobile-Android-Landscape-linux.png:1]: Thanks, fixed

File: packages/react-composites/src/composites/common/Drawer/MoreDrawer.tsx (hunk 9 above)
Comment (rzdor) [reply packages/react-composites/src/composites/common/Drawer/MoreDrawer.tsx]: changed

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/ClosedCaptions.test.ts-snapshots/caption-button-teams-call-Mobile-Android-Landscape-linux.png
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/ClosedCaptions.test.ts-snapshots/caption-button-teams-call-Mobile-Android-Landscape-linux.png:1]: Just a note for posterity that this captions button test no longer captures the captions button as its pushed off screen. Don't think there's much to do though here

**Review by JamesBurnside — APPROVED** [review]

**Review by dmceachernmsft — APPROVED** [review]

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**